src/index.ts

```diff
@@ -132,87 +149,103 @@ class LiskMigrator extends Command {
+				const { version: appVersion } = nodeInfo;
 
-			if (autoStartLiskCoreV4) {
-				if (!networkConstant) {
+				cli.action.start('Verifying if backup height from node config matches snapshot height');
+				if (snapshotHeight !== nodeInfo.backup.height) {
 					this.error(
-						`Unknown network detected. No NETWORK_CONSTANT defined for networkID: ${networkIdentifier}.`,
+						`Lisk Core v3 backup height (${nodeInfo.backup.height}) does not match the expected snapshot height (${snapshotHeight}).`,
 					);
 				}
-			}
+				cli.action.stop('Snapshot height matches backup height');
 
-			cli.action.start('Verifying Lisk Core version');
-			const isLiskCoreVersionValid = semver.valid(appVersion);
-			if (isLiskCoreVersionValid === null) {
-				this.error(
-					`Invalid Lisk Core version detected: ${appVersion}. Minimum supported version is ${MIN_SUPPORTED_LISK_CORE_VERSION}.`,
+				cli.action.start(
+					`Verifying snapshot height to be multiples of round length i.e ${ROUND_LENGTH}`,
 				);
-			}
+				if (snapshotHeight % ROUND_LENGTH !== 0) {
+					this.error(
+						`Invalid snapshot height provided: ${snapshotHeight}. It must be an exact multiple of round length (${ROUND_LENGTH}).`,
+					);
+				}
+				cli.action.stop('Snapshot height is valid');
 
-			// Using 'gt' instead of 'gte' because the behavior is swapped
-			// i.e. 'gt' acts as 'gte' and vice-versa
-			if (semver.gt(`${MIN_SUPPORTED_LISK_CORE_VERSION}-rc.0`, appVersion)) {
-				this.error(
-					`Lisk Migrator is not compatible with Lisk Core version ${appVersion}. Minimum supported version is ${MIN_SUPPORTED_LISK_CORE_VERSION}.`,
-				);
-			}
-			cli.action.stop(`${appVersion} detected`);
+				// Asynchronously capture the node's Forging Status information at the snapshot height
+				// This information is necessary for the node operators to enable generator post-migration without getting PoM'd
+				captureForgingStatusAtSnapshotHeight(this, client, snapshotHeight, outputDir);
+
+				if (autoStartLiskCoreV4) {
+					if (!networkConstant) {
+						this.error(
+							`Unknown network detected. No NETWORK_CONSTANT defined for networkID: ${networkIdentifier}.`,
+						);
+					}
+				}
 
-			// User specified custom config file
-			const configV3: ApplicationConfigV3 = customConfigPath
-				? await getConfig(this, liskCoreV3DataPath, networkIdentifier, customConfigPath)
-				: await getConfig(this, liskCoreV3DataPath, networkIdentifier);
+				cli.action.start('Verifying Lisk Core version');
+				const isLiskCoreVersionValid = semver.valid(appVersion);
+				if (isLiskCoreVersionValid === null) {
+					this.error(
+						`Invalid Lisk Core version detected: ${appVersion}. Minimum supported version is ${MIN_SUPPORTED_LISK_CORE_VERSION}.`,
+					);
+				}
+
+				// Using 'gt' instead of 'gte' because the behavior is swapped
+				// i.e. 'gt' acts as 'gte' and vice-versa
+				if (semver.gt(`${MIN_SUPPORTED_LISK_CORE_VERSION}-rc.0`, appVersion)) {
```

Comment:
I am not sure if we should take rc has the min supported? we should take the release that had all the fixes in this case `> 3.1.0` I guess or `3.1.1`